contracts/minter-suite/Minters/MinterHolder/MinterHolderV2.sol

```diff
@@ -0,0 +1,765 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../../interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "../../../interfaces/0.8.x/IMinterFilterV0.sol";
+import "../../../interfaces/0.8.x/IFilteredMinterHolderV0.sol";
+import "../../../interfaces/0.8.x/IDelegationRegistry.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin-4.5/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin-4.5/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin-4.5/contracts/utils/structs/EnumerableMap.sol";
+
+pragma solidity 0.8.17;
+
+/**
+ * @title Filtered Minter contract that allows tokens to be minted with ETH
+ * when purchaser owns an allowlisted ERC-721 NFT.
+ * This is designed to be used with IGenArt721CoreContractV3 contracts.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with limited powers.
+ * Privileged roles and abilities are controlled by the core contract's Admin
+ * ACL contract and a project's artist. Both of these roles hold extensive
+ * power and can modify minter details.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to the core contract's Admin ACL
+ * contract:
+ * - registerNFTAddress
+ * - unregisterNFTAddress
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to a project's artist:
+ * - allowHoldersOfProjects
+ * - removeHoldersOfProjects
+ * - allowRemoveHoldersOfProjects
+ * - updatePricePerTokenInWei
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ */
```

Comment:
Let's expand on the file comment with something about the vault-delegate functionality, similar to what we have here in the MerkleMinter: https://github.com/ArtBlocks/artblocks-contracts/blob/main/contracts/minter-suite/Minters/MinterMerkle/MinterMerkleV2.sol#L36-L48

🙏

Reply:
Done and done! I updated the language to reflect that this contract only allows delegating at the token level and not the contract.

Reply:
So contract-level delegations are still possible and supported by this minter–they get inherited when we perform a token-specific lookup check. I think we may want to adjust wording here slightly and perhaps reword the language a bit in https://github.com/ArtBlocks/artblocks-contracts/blob/main/contracts/minter-suite/Minters/MinterMerkle/MinterMerkleV2.sol#L36-L48 as well now that I'm looking at it again, as I realizing that it can be misunderstood; cc @ryley-o

Reply:
@jakerockland gaaaa yeah idk what I was thinking when writing those lines in the Merkle Minter.
Merkle V2 should say contract-level or wallet-level delegations are alllowed, and this minter should say token-level, contract-level, or wallet-level delegations are allowed.

Reply:
I made a few more tweaks just to be really clear about what our minters do/don't handle as far as the delegation goes. But please feel free to tear these up if I'm offbase with any of my wording 😅 
https://github.com/ArtBlocks/artblocks-contracts/pull/357/commits/a01d69d00f0be4b837c6a2218db853af2a254a46